gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java

```diff
@@ -237,17 +258,31 @@ public EmailWrapper createEmail(io.gravitee.am.model.Template template, Client c
 
         // compute email subject
         final Template plainTextTemplate = new Template("subject", new StringReader(emailTpl.getSubject()), freemarkerConfiguration);
-        email.setSubject(processTemplateIntoString(plainTextTemplate, params));
+        email.setSubject(processTemplate(plainTextTemplate, params, preferredLanguage));
 
         // compute email content
         final Template subjectTemplate = freemarkerConfiguration.getTemplate(email.getTemplate());
-        email.setContent(processTemplateIntoString(subjectTemplate, params));
+        email.setContent(processTemplate(subjectTemplate, params, preferredLanguage));
 
         EmailWrapper wrapper = new EmailWrapper(email);
         wrapper.setExpireAt(expiresAt);
         return wrapper;
     }
 
+    private String processTemplate(Template plainTextTemplate, Map<String, Object> params, Locale preferredLanguage) throws TemplateException, IOException {
+        StringWriter result = new StringWriter(1024);
+
+        Map entries = new HashMap(params);
```

Comment:
Missing type params and `<>`.
```suggestion
        Map<Object, Object> entries = new HashMap<>(params);
```